### DISCUSSION TITLE
Passing dictionary as labels in python - prometheus

### DISCUSSION THREAD

**Discussion by Yamunaindiran** [prometheus/client_python#713] [Q&A] [answered]
gdiskc = Gauge("disk_usage_c_drive", "Disk Usage C Drive", disk_usageC.keys())
gdiskd = Gauge("disk_usage_d_drive", "Disk Usage D Drive", disk_usageD.keys())

def set_calc_():
   gdiskc.labels(**disk_usageC)
    gdiskd.labels(**disk_usageD)

Metrics looks like this:
disk_usage_c_drive{free="61112762368",percent="52.6",total="128849014784",used="67736252416"} **0.0**

I need to get rid of the 0.0 in disk_usage_c_drive gauge as my graph is grafana is 0. 

I'm trying to write windows exporter in python. please help.

**Comment by csmarchbanks** [comment, marked as answer]
Hello, I converted this from an issue to a discussion as it appears to be a usage question.

You should not be including the value of a metric as labels such as in your example. Every time your disk usage changes right now Prometheus will create a new series which will be very expensive and hard to use. Typically you will want multiple metrics each with the value. For example:
```
disk_usage_free_bytes{drive="c"} 61112762368
disk_usage_total_bytes{drive="c"} 128849014784
disk_usage_used_bytes{drive="c"} 67736252416
disk_usage_used_ratio{drive="c"} 0.526
```

Does that help? You can take a look at some of the metrics in the windows exporter to get an idea: https://github.com/prometheus-community/windows_exporter/blob/master/docs/collector.logical_disk.md.

**Reply by Yamunaindiran** [reply]
Yes, your explanation is right, I rectified the above by:
gdiskc = Gauge("disk_usage_c_drive", "Disk Usage C Drive", ['disk_type_c'])
gdiskc.labels("total").set(disk_usageC.total)
    gdiskc.labels("free").set(disk_usageC.free)
    gdiskc.labels("used").set(disk_usageC.used)
    gdiskc.labels("percent").set(disk_usageC.percent)


This works fine now. Thank you for answering.
